### PR TITLE
trace: split out trace-soflogger.h and trace-boot.h

### PR DESCRIPTION
**Pull request by kv2019i**
The SOF trace.h is a bit hard to follow as it mixes frontend and backend definitions. The frontend definitions like tr_info() are needed by all SOF builds. The backends can vary and the application traces can be routed to SOF DMA trace (sof-logger), Zephyr logging subsystem or optionally compiled out when tracing is disabled.

Improve readability of the header by moving all definitions related to the sof-logger backend to a separate trace-soflogger.h header file.

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST
